Tools/LyTestTools/ly_test_tools/cli/codeowners_hint.py

```diff
@@ -0,0 +1,232 @@
+            # glob remainders of A against remainder of B
+            for target_index in pre_wildcard_target_end_indices:  # may be multiple substring matches within target
+                target_remainder = target_str[target_index:]
+                if pathlib.PurePosixPath(target_remainder).match(wildcard_with_remainder_owned):
+                    matched = True
+                    break  # exit early on success
+    else:  # simple path matching
+        if owned_path.startswith("/"):  # ownership of specific directory: verify if A exists inside B
+            try:  # is_relative_to introduced in Python 3.9
+                target_path.relative_to(owned_path)
+                matched = True
+            except ValueError:
+                matched = False
+        else:  # ownership of all relative directories: verify if B is a substring of A
+            matched = owned_path in str(target_path)
+    return matched
+
+
+def _pretty_print_success(print_fn, found_codeowners_path, matched_path, owner_aliases):
+    """
+    :param print_fn: function to call when logging strings
+    :param found_codeowners_path: verified path to a GitHub CODEOWNERS file
+    :param matched_path: first part of an entry matched in the CODEOWNERS file
+    :param owner_aliases: second part of an entry in a CODEOWNERS file
+    """
+    print_fn(f"Matched '{matched_path}' in file {found_codeowners_path}")
+    print_fn(f"For additional support please reach out to: {owner_aliases}")
+
+
+def _pretty_print_failure(print_fn, found_codeowners_path, matched_path, original_target,
+                          default_alias=_DEFAULT_CODEOWNER_ALIAS):
+    """
+    :param print_fn: function to call when logging strings
+    :param found_codeowners_path: verified path to a GitHub CODEOWNERS file which, empty when missing
+    :param matched_path: entry matched in the CODEOWNERS file, empty when not matched
+    :param original_target: the path which matching was attempted on
+    :param default_alias: who to contact as no owner was found
+    """
+    if not found_codeowners_path:
+        print_fn(f"No GitHub CODEOWNERS file was found for '{original_target}'")
+    else:
+        if not matched_path:
+            print_fn(f"No ownership information for '{original_target}' found in file {found_codeowners_path}")
+        else:
+            print_fn(f"Ownership for '{matched_path}' is explicitly empty in file {found_codeowners_path}")
+    print_fn(f"For additional support please reach out to: {default_alias}")
+
+
+def _main() -> int:
+    parser = argparse.ArgumentParser(description="Display GitHub CODEOWNERS information for a path to stdout")
+    parser.add_argument('target', metavar='T',
+                        help="file path to find an owner for")
+    parser.add_argument('-c', '--codeowners',
+                        help="path to a GitHub CODEOWNERS file, when not set it will be searched for in the repo "
+                             "containing the target")
```

Comment:
Adding a type keyword here allows, us to enforce that the args.codeowners is a `pathlib.Path`